src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -0,0 +1,346 @@
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Module providing a transformation from an Assignment node
+containing an Array Reference node in its left-hand-side which in turn
+has at least one PSyIR Range node specifying an access to an array
+index (equivalent to an array assignment statement in Fortran) to the
+equivalent loop representation using a NemoLoop node. A Range node is
+provided to the apply method of the tranformation to indicate which
+array index should be transformed.
+
+'''
+
+from __future__ import absolute_import
+
+from psyclone.psyGen import Transformation
+from psyclone.psyir.transformations.transformation_error \
+    import TransformationError
+from psyclone.errors import InternalError
+from psyclone.psyir.symbols import DataSymbol, INTEGER_TYPE
+from psyclone.psyir.nodes import Range, Reference, ArrayReference, \
+    Assignment, Literal, Operation, BinaryOperation
+from psyclone.nemo import NemoLoop, NemoKern
+from psyclone.configuration import Config
+
+
+class NemoArrayRange2LoopTrans(Transformation):
+    '''Provides a transformation from a PSyIR ArrayReference Range to a
+    PSyIR NemoLoop. For example:
+
+    >>> from psyclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "nemo"
+    >>> filename = "tra_adv.F90" # examples/nemo/code
+    >>> ast, invoke_info = parse(filename, api=api)
+    >>> psy = PSyFactory(api).create(invoke_info)
+    >>> schedule = psy.invokes.invoke_list[0].schedule
+    >>> schedule.view()
+    >>>
+    >>> from psyclone.psyir.nodes import Range
+    >>> from psyclone.domain.nemo.transformations import \
```

Comment:
Picky (might as well start the year as I mean to go on) but you don't need the ">>>" after the line-continuation symbol here as it makes for strangely-rendered code in the User Guide.

Reply:
I don't think that is picky :-)